### PR TITLE
feat(auth): allow configurable expiration time

### PR DESCRIPTION
**Pull request by QisFj**
**Description**

:rotating_light: Before submitting your PR, please read [community](https://github.com/filebrowser/community), and indicate which issues (in any of the repos) are either fixed or closed by this PR. See [GitHub Help: Closing issues using keywords](https://help.github.com/articles/closing-issues-via-commit-messages/).

- [x] DO make sure you are requesting to **pull a topic/feature/bugfix branch** (right side). Don't request your master!
- [x] DO make sure you are making a pull request against the **master branch** (left side). Also you should start *your branch* off *our master*.
- [x] DO make sure that File Browser can be successfully built. See [builds](https://github.com/filebrowser/community/blob/master/builds.md) and [development](https://github.com/filebrowser/community/blob/master/development.md).
- [x] DO make sure that related issues are opened in other repositories. I.e., the frontend, caddy plugins or the web page need to be updated accordingly.
- [x] AVOID breaking the continuous integration build.

**Further comments**

just another implementation to https://github.com/filebrowser/filebrowser/pull/1794

### CODE REVIEW

File: settings/settings.go [hunk 1]
```diff
@@ -64,3 +67,49 @@ func GenerateKey() ([]byte, error) {
 
 	return b, nil
 }
+
+type Duration time.Duration // support json Marshal/Unmarshal for time.Duration
+
+func (dur Duration) MarshalJSON() ([]byte, error) {
+	return []byte("\"" + time.Duration(dur).String() + "\""), nil
```
Comment (o1egl) [inline settings/settings.go]:
```suggestion
	return []byte(strconv.Quote(time.Duration(dur).String()), nil
```

File: settings/settings_test.go [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+				midStr, err := codec.marshal(dur)
+				require.NoError(t, err)
+				err = codec.unmarshal(midStr, &dur)
+				require.NoError(t, err)
+				require.Equal(t, tc.value, dur)
+			})
+		}
+	}
+}
```
Comment (o1egl) [inline settings/settings_test.go]:
```suggestion
	for _, tc := range testCases {
		t.Run(tc.srt, func(t *testing.T){
			for _, codec := range codecs {
				t.Run(codec.name, func(t *testing.T) {
					// str --> dur --> mid_str(may different from str) --> dur
					var dur Duration
					err := codec.unmarshal([]byte(tc.str), &dur)
					require.NoError(t, err)
					require.Equal(t, tc.value, due)

					midStr, err := codec.marshal(dur)
					require.NoError(t, err)
					err = codec.unmarshal(midStr, &dur)
					require.NoError(t, err)
					require.Equal(t, tc.value, dur)
				})
			}
		})
	}
}
```

**Review by o1egl — COMMENTED** [review]
Added some suggestions. Looks good other than that

**Comment by QisFj** [comment]
followed your suggestions. and renamed `TokenExpirationTime` to `DefaultTokenExpirationTime`

**Comment by ados8** [comment]
Would be amazing to have this feature implemented, is it still on the roadmap. 😊

**Comment by matatentacle** [comment]
+1 Much needed for transfer of multiple video files.

**Comment by mcmartins** [comment]
+1

**Comment by m4heshd** [comment]
Why do PRs related to this feature always get ignored?

**Comment by ados8** [comment]
I think the project is no longer being maintained.
It's overkill but look into Nextcloud.

**Comment by m4heshd** [comment]
> Nextcloud

I already have an instance up and yeah, way too overkill to replace Filebrowser.

**Comment by alanmilinovic** [comment]
+1

**Comment by Mysterius** [comment]
+1 here too. A shame that this apparently almost ready PR was let down.

**Pull request closed without merging**